packages/cozy-harvest-lib/test/components/TriggerLauncher.spec.js

```diff
@@ -239,14 +251,43 @@ describe('TriggerLauncher', () => {
         </TriggerLauncher>
       )
 
-      wrapper.instance().launch()
+      wrapper.instance().launch(trigger)
       wrapper.instance().displayTwoFAModal()
       await wrapper.instance().handleSuccess()
       wrapper.update()
       expect(wrapper.getElement()).toMatchSnapshot()
     })
   })
 
+  describe('handleLoginSuccess', () => {
+    it('should hide twoFA modal', async () => {
+      const wrapper = shallow(
+        <TriggerLauncher {...props}>
+          {({ launch, running }) => <Child launch={launch} running={running} />}
+        </TriggerLauncher>
+      )
+
+      wrapper.instance().launch(trigger)
+      wrapper.instance().displayTwoFAModal()
+      await wrapper.instance().handleLoginSuccess()
+      wrapper.update()
+      expect(wrapper.getElement()).toMatchSnapshot()
+    })
+
+    it('should call the onLoginSuccess callback', () => {
+      const onLoginSuccess = jest.fn()
+      const wrapper = shallow(
+        <TriggerLauncher {...props} onLoginSuccess={onLoginSuccess}>
+          {({ launch, running }) => <Child launch={launch} running={running} />}
+        </TriggerLauncher>
+      )
+
+      wrapper.instance().launch(trigger)
+      wrapper.instance().handleLoginSuccess()
+      expect(onLoginSuccess).toHaveBeenCalled()
+    })
+  })
```

Comment:
since `TriggerLauncher` is now retuning also a `trigger` should we add a test for it?

Reply:
Also, since we don't call `handleLoginSuccess` by hand in our code, can we add a test for our real case? ie dispatching the event and check if `handleLoginSuccess` is called or something similar?

Reply:
It looks like testing the event dispatching library we are using here, MicroEE. Maybe we should test KonnectorJob instead.

Reply:
This is not the implementation of the event I want to test. I just want to test, that if the flow goes right, we call `handleLoginSuccess` automatically. So yeah, it's basically KonnectorJob